src/controller/app_routes.rs

```diff
@@ -178,6 +179,12 @@ impl AppRoutes {
             }
         }
 
+        if let Some(compression) = &ctx.config.server.middlewares.compression {
```

Comment:
Not related/limited to this PR, but does `AppRoutes` really need to know about all available middlewares? What about future custom middlewares that aren't part of loco?

Reply:
That's a really great question. Actually we really need more of those questions for deliberating around opinions/decisions!

The idea is that we have a set of "blessed" middleware that are strongly tied, and now we can say -- which are the ones that are really in say 80% of projects (just 80/20) rule to include, also assuming they "cost" us very little (in binary size, dependencies, and maintenance) but provide the user maximum value.

We want to keep the list very small as you say. We also assume compression is such a middleware.
What are your thoughts specifically? now that you have the context, do you think adding compression as a "first-class" is too heavy?